### PR TITLE
CB-18053 Fix failing Jackson serialization

### PR DESCRIPTION
**Pull request by bergerdenes**
found in the nightly E2E tests' logs.
The most interesting was fixing `com.sun.proxy.$Proxy`-typed interface
(InstanceMetadataView).
Adjusted the serializability checker too.

### CODE REVIEW

File: environment-api/src/main/java/com/sequenceiq/environment/api/v1/environment/model/EnvironmentNetworkGcpParams.java [hunk 1]
```diff
@@ -45,15 +45,15 @@ public void setSharedProjectId(String sharedProjectId) {
...
+    public Boolean isNoPublicIp() {
         return noPublicIp;
     }
 
     public void setNoPublicIp(Boolean noPublicIp) {
         this.noPublicIp = noPublicIp;
     }
 
-    public Boolean getNoFirewallRules() {
+    public Boolean isNoFirewallRules() {
```
Comment (lajosrodek) [inline environment-api/src/main/java/com/sequenceiq/environment/api/v1/environment/model/EnvironmentNetworkGcpParams.java:56]: Like above, the `getX()` naming is the appropriate one due to the usage of a reference type.

**Review by lajosrodek — CHANGES_REQUESTED** [review]

File: environment-api/src/main/java/com/sequenceiq/environment/api/v1/environment/model/EnvironmentNetworkAzureParams.java [hunk 2]
```diff
@@ -48,7 +48,7 @@ public void setResourceGroupName(String resourceGroupName) {
         this.resourceGroupName = resourceGroupName;
     }
 
-    public Boolean getNoPublicIp() {
+    public Boolean isNoPublicIp() {
```
[the remark this answers is not in the document]
Comment (bergerdenes) [reply environment-api/src/main/java/com/sequenceiq/environment/api/v1/environment/model/EnvironmentNetworkAzureParams.java]: fixing checker, and renaming Boolean getters to getXX()

File: freeipa/src/main/java/com/sequenceiq/freeipa/entity/Stack.java [hunk 3]
```diff
@@ -223,7 +223,7 @@ public void setGatewayport(Integer gatewayport) {
         this.gatewayport = gatewayport;
     }
 
-    public boolean isUseCcm() {
+    public Boolean getUseCcm() {
         return Boolean.TRUE.equals(useCcm);
     }
```
Comment (lajosrodek) [inline freeipa/src/main/java/com/sequenceiq/freeipa/entity/Stack.java:228]: This is not a pure getter and cannot ever return `null`, so perhaps the original signature was more appropriate. (But then a new method is needed for the real getter.)

File: freeipa/src/main/java/com/sequenceiq/freeipa/entity/Stack.java [hunk 4]
```diff
@@ -239,7 +239,7 @@ public void setTunnel(Tunnel tunnel) {
         this.tunnel = tunnel;
     }
 
-    public boolean isClusterProxyRegistered() {
+    public Boolean getClusterProxyRegistered() {
         return Boolean.TRUE.equals(clusterProxyRegistered);
     }
```
Comment (lajosrodek) [inline freeipa/src/main/java/com/sequenceiq/freeipa/entity/Stack.java:244]: LIke above, this is not a pure getter, so the original signature was more appropriate.

**Review by lajosrodek — CHANGES_REQUESTED** [review]

File: freeipa/src/main/java/com/sequenceiq/freeipa/entity/Stack.java (hunk 3 above)
Comment (bergerdenes) [reply freeipa/src/main/java/com/sequenceiq/freeipa/entity/Stack.java:228]: I don't wanna. this is a very rare edge case (2 occurrences) and would unnecessarily complicate the checker logic.

File: freeipa/src/main/java/com/sequenceiq/freeipa/entity/Stack.java (hunk 4 above)
Comment (bergerdenes) [reply freeipa/src/main/java/com/sequenceiq/freeipa/entity/Stack.java:244]: see above

**Review by lajosrodek — APPROVED** [review]
LGTM, the above two remarks are not critical and would be complicated to handle in the serializability checker.

**Pull request merged**